lib/booking.rb

```diff
@@ -0,0 +1,81 @@
+require 'pry'
+require 'time'
+require 'date'
+
+module Hotel
+  class BookingSystem
+    attr_reader :rooms, :reservations
+
+    def initialize
+      # times loop to create an array of 20 rooms
+      @rooms = []
+      20.times do |room_num|
+        room_num += 1
+        @rooms << Room.new(room_num)
+      end
+      @reservations = []
+    end
+
+    # method to find available rooms and return an array of available room
+    def find_rooms_available(start_date, end_date)
+
+      rooms_available = []
+
+      reserv_range = DateRange.new(start_date, end_date)
+      x = reserv_range.get_date_range
+
+      @rooms.each do |room|
+        if room.room_reservations.empty?
+          rooms_available << room
+        elsif
+          room.room_reservations.each do |reservation|
+            if reservation.date_range.check_overlap(x) == true
+              rooms_available << room
+            end
+          end
+        end
+      end
+      return rooms_available
+    end
+
+    # list the reservations for a specific date
+    def list_reservations(date)
+
+      return @reservations.select do |reservation|
+        reservation.date_range.check_include?(date)
+      end
+    end
+
+
+    def make_reservation(start_date, end_date)
+      # find a room that is available
+      room_to_book = find_rooms_available(start_date, end_date).first
+
+      if room_to_book == nil
+        raise StandardError.new ("No rooms available")
+      else
+        info = {id: rand(1..300),
```

Comment:
Because the `if` clause raises an error, you don't need to wrap the rest of the method in an `else`. You could instead say:

```ruby
if room_to_book == nil
  raise StandardError.new ("No rooms available")
end
info = { # ...
```